utils/modules/polygon_utils.py

```diff
@@ -177,27 +178,38 @@ def pols_neighbor_num(verts, pols):
     '''
     return [len(p) for p in pols_neighbor(verts, pols)]
 
-def pols_normals(vertices, faces):
+def pols_normals(vertices, faces, output_numpy):
     '''
     Returns Faces normals as [vector, vector,...]
     vertices: list as [vertex, vertex, ...], being each vertex [float, float, float].
     faces: list as [polygon, polygon,..], being each polygon [int, int, ...].
     '''
-    bm = bmesh_from_pydata(vertices, [], faces, normal_update=True)
-    vals = [tuple(face.normal) for face in bm.faces]
-    bm.free()
+    vals = np_process_polygons(vertices, faces, func=np_faces_normals, output_numpy=output_numpy)
     return vals
 
-def pols_absolute_normals(vertices, faces):
+def np_faces_normals(v_pols):
+    pol_sides = v_pols.shape[1]
+    if pol_sides > 3:
+        f_normals = np.zeros((len(v_pols), 3), dtype=np.float64)
+        for i in range(pol_sides - 2):
+            f_normals += np.cross(v_pols[::, (1+i)%pol_sides] - v_pols[::, 0], v_pols[::, (2+i)%pol_sides] - v_pols[::, 0])
```

Comment:
this is not expecting ngons / irregular / concavity ?

Reply:
Actually it  does react very similar to concave ngons to the standard bmesh calculation
![image](https://user-images.githubusercontent.com/10011941/114272579-0abcc200-9a17-11eb-841f-7a221b912b91.png)

Reply:
most of the geometry produced by sverchok will go through this no problem, but for complex stuff we need a tessellator of the ngon.

if society wasn't being actively phucked as we speak i would invest more time in that myself:)

Reply:
What kind of precision can someone expect from the normal calculation of a non-planar concave polygon? :)

Reply:
i'm talking about planar concave polygons. I don't see how the above code will work for those, some normals will be flipped .

Reply:
it will be fine for most applications, but this is why the VDdraw has to tesselate ngons first before assigning normal derived colours.

Reply:
i'd be happy to completely misunderstand this :)

Reply:
as i understand it the correct tesselation is only necessary to display the resulting triangles not to compute de normals

Reply:
![image](https://user-images.githubusercontent.com/10011941/114275000-2af17e80-9a21-11eb-8429-d9cebcd990f8.png)

Reply:
ifso, then inset_specials can be done much faster too

Reply:
yes, have it almost ready. faster and with more options :D

Reply:
well that's very cool! 
tested this now. I can't break it changed to seed in `ngon` :) , disregarding the invalid ngons ofcourse.

![image](https://user-images.githubusercontent.com/619340/114284645-8c314600-9a51-11eb-9474-5bcadf0e1eac.png)

[ngon test_2021_04_10_21_09.zip](https://github.com/nortikin/sverchok/files/6290767/ngon.test_2021_04_10_21_09.zip)

nice work.

Reply:
Thanks @zeffii for  understanding the joy of a  Numpy Normals algorithm.. :D